### PR TITLE
gui-apps/waybar: add several USE flags

### PR DESCRIPTION
**Pull request by MrRoy**
evdev, jack, libinput, logind, mpris, pipewire, systemd, test, upower
drop net-wireless/rfkill DEPEND

Closes: https://bugs.gentoo.org/894068

### CODE REVIEW

File: gui-apps/waybar/waybar-0.9.17-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	>=dev-libs/spdlog-1.8.5:=
+	dev-libs/date:=
+	dev-libs/wayland
+	gui-libs/wlroots:=
+	x11-libs/gtk+:3[wayland]
+	x11-libs/libxkbcommon
+	evdev? ( dev-libs/libevdev:= )
+	jack? ( virtual/jack )
+	libinput? ( dev-libs/libinput:= )
+	logind? ( || ( sys-apps/systemd sys-auth/elogind ) )
```
Comment (juippis) [inline gui-apps/waybar/waybar-0.9.17-r1.ebuild:49]: Could you multi-line this? It's a bit hard to read. 

What does `systemd` use flag do that isn't covered by this? I feel like there's some combining that needs to be done, or if `+systemd` will **just** install service files, then always install those unconditionally.

File: gui-apps/waybar/waybar-0.9.17-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	mpd? ( media-libs/libmpdclient )
+	mpris? ( >=media-sound/playerctl-2 )
+	network? ( dev-libs/libnl:3 )
+	popups? ( gui-libs/gtk-layer-shell )
+	pipewire? ( media-video/wireplumber:0/0.4 )
+	pulseaudio? ( media-sound/pulseaudio )
```
Comment (juippis) [inline gui-apps/waybar/waybar-0.9.17-r1.ebuild:55]:
```suggestion
	pulseaudio? ( media-sound/pulseaudio[daemon] )
```

File: gui-apps/waybar/metadata.xml [hunk 3]
```diff
@@ -15,10 +15,15 @@
 	</maintainer>
 	<use>
 		<flag name="experimental">Enable experimental features</flag>
```
Comment (juippis) [inline gui-apps/waybar/metadata.xml:17]: Such as? :) are there any examples that this holds.

**Review by juippis — COMMENTED** [review]
Looks good, just some nitpicks mostly. This is starting enter a territory where we may not want to add any more use flags, but then again the description does say "highly customizable".

File: gui-apps/waybar/metadata.xml (hunk 3 above)
Comment (MrRoy) [reply gui-apps/waybar/metadata.xml]: I added a short example to clarify

File: gui-apps/waybar/waybar-0.9.17-r1.ebuild (hunk 1 above)
Comment (MrRoy) [reply gui-apps/waybar/waybar-0.9.17-r1.ebuild]: For the systemd USE flag: you are correct, its only purpose is to decide whether to install the systemd unit file. Unfortunately the meson.build file checks for the presence of systemd when this option is enabled, and fetches the unit dir path, so I included a patch that hardcodes the path to `/usr/lib/systemd/user` and removes the check in meson.build.

For the logind USE flag, its purpose is to enable/disable the idle inhibit module, which requires (e)logind

Comment (juippis) [reply gui-apps/waybar/waybar-0.9.17-r1.ebuild]: Thanks for the effort but let's not overcomplicate things. This patch will make future maintenance harder, unless it is accepted and merged upstream. Then we can obviously add it. 

If it checks that systemd exists, then the use flag makes more sense.

**Comment by thesamesam** [comment]
Ugh, sorry, I've merged this but I squashed and forgot to add the tag.

File: gui-apps/waybar/waybar-0.9.17-r1.ebuild (hunk 1 above)
Comment (MrRoy) [reply gui-apps/waybar/waybar-0.9.17-r1.ebuild]: @juippis since it's already merged, I'll take a note to remove the patch for the next update (eg. 0.9.18)

**Pull request closed without merging**